2016/proposals/chris-medrela.md

```diff
@@ -0,0 +1,461 @@
+
+1. [Milestone 1: Managing SWC Checkout Process](#milestone-1-managing-swc-checkout-process) -- that is Homework, Discussion Session, Checkout Session and welcoming to the community.
+
+1. [Milestone 2: Managing DC Checkout Process](#milestone-2-managing-dc-checkout-process)
+
+1. [Milestone 3: Managing Recruiting Candidates](#milestone-3-managing-recruiting-candidates) and rest of the workflow, that is matching them to Instructor Trainings and running those trainings.
+
+It may be necessary to move from SQLite to Postgress. I'm discussing in [#716](https://github.com/swcarpentry/amy/issues/716) with Piotr Banaszkiewicz if it's really necessary, but if it is, it must be done at the beginning, as an additional milestone.
+
+I believe it's really important to get feedback as soon as possible, therefore:
+
+- Before introducing new features to all users, they'll be tested by a group of volunteering early users. That way, we don't have to test everything before showing it to first users, so we can get feedback faster and iterate quicker.
+
+- The early users can give feedback in an easy way, e.g. by sending an email rather than opening an issue on github.
+
+- UI mockups will be presented and accepted by the early users.
+
+- Amy dev instance will be available to all interested parties.
+
+It's also important to note what will be **not** implemented:
+
+- Advanced security control. There'll be only one flag in `Person` model that indicates if somebody is a Trainee with very limited permissions or somebody else (Admin, Trainer etc.) who can can edit everything.
+
+- Responsive interface.
+
+- Tracking sessions with Google Analytics or Piwik.
+
+## Milestone 1: Managing SWC Checkout Process
+
+That is everything after an Instructor Training -- that is Homework, Discussion Session, Checkout Session and welcoming to the community.
+
+The new workflow would be:
+
+- At the end of Instructor Training, Trainees register in Amy using the link shared by the Trainer. The link would be unique for each Training, so Trainees can be automatically associated with that Training.
+
+- The Amy interface for new trainees would be completely different from the current one. It could be a single page for the entire Checkout Process, plus one form to update their profile and one to change password.
+
+- Trainees make a change in the SWC Lessons and paste the link to their pull requests into Amy.
+
+- Amy analyzes the link to decide which core lesson did the trainee pick up.
+
+- After entering the pull request, the trainee uses Amy to register for one Discussion Session.
+
+    - In order not to overload Lesson Maintainers, they don't have to indicate in Amy if the pull request is good enough. It also means that trainees don't have to wait for the feedback from lesson maintainers, so they can register for discussion session immediately. This may raise up the percent of trainees who bother up to go through rest of checkout process.
```

Comment:
I agree that trainers can/should do this manually for the first version, 
but for later versions, it would be handy if we could harvest 
information from GitHub (e.g., look at tags attached to the trainees' 
PRs) and automatically update the status in AMY.